ipa/ipa_azure.py

```diff
@@ -82,176 +82,450 @@ def __init__(self,
+                subnet_name,
+                {'address_prefix': '10.0.0.0/29'}
+            )
+        except Exception as error:
+            raise AzureProviderException(
+                'Unable to create subnet: {0}.'.format(error)
+            )
 
-    def _get_virtual_machine(self):
-        """Return instance of VirtualMachine class."""
-        return VirtualMachine(self.account)
+        return subnet_operation.result()
 
-    def _is_instance_running(self):
+    def _create_virtual_network(self, region, resource_group_name, vnet_name):
         """
-        Return True if instance is in running state.
+        Create a vnet in the given resource group with default address space.
+        """
+        try:
+            vnet_operation = self.network.virtual_networks.create_or_update(
+                resource_group_name,
+                vnet_name,
+                {
+                    'location': region,
+                    'address_space': {
+                        'address_prefixes': ['10.0.0.0/27']
+                    }
+                }
+            )
+        except Exception as error:
+            raise AzureProviderException(
+                'Unable to create vnet: {0}.'.format(error)
+            )
 
-        Raises:
-            AzureProviderException: If state is Undefined.
+        vnet_operation.wait()
+
+    def _create_vm_parameters(self, interface):
         """
-        state = self._get_instance_state()
+        Create the VM parameters dictionary.
 
-        if state == 'Undefined':
+        Requires an existing network interface object.
+        """
+        # Split image ID into it's components.
+        self._process_image_id()
+
+        return {
+            'location': self.region,
+            'os_profile': {
+                'computer_name': self.running_instance_id,
+                'admin_username': self.ssh_user,
+                'linux_configuration': {
+                    'disable_password_authentication': True,
+                    'ssh': {
+                        'public_keys': [{
+                            'path': '/home/{0}/.ssh/authorized_keys'.format(
+                                self.ssh_user
+                            ),
+                            'key_data': self.ssh_public_key
+                        }]
+                    }
+                }
+            },
+            'hardware_profile': {
+                'vm_size': self.instance_type or AZURE_DEFAULT_TYPE
+            },
+            'storage_profile': {
+                'image_reference': {
+                    'publisher': self.image_publisher,
+                    'offer': self.image_offer,
+                    'sku': self.image_sku,
+                    'version': self.image_version
+                },
+            },
+            'network_profile': {
+                'network_interfaces': [{
+                    'id': interface.id,
+                    'primary': True
+                }]
+            }
+        }
```

Comment:
There's is really nothing wrong with creating temporary variables. This will make the composition of the information much more digestible.